xml/1.4/modules/ROOT/pages/xml-module-troubleshooting.adoc

```diff
@@ -0,0 +1,111 @@
+********************************************************************************
+Message               : The supplied schemas were not valid. schema_reference: Failed to read schema document 'schemadocumentname.xsd', because 'file' access is not allowed due to restriction set by the accessExternalSchema property.
+Element               : bug-xsd-schema-not-foundFlow/processors/2 @ bug-xsd-schema-not-found-support:main.xml:16 (Validate schema)
+Element DSL           : <xml-module:validate-schema schemas="ref/despatchAdviceMessages/despatchAdviceMessage.xsd"></xml-module:validate-schema>
+Error type            : XML-MODULE:INVALID_SCHEMA
+FlowStack             : at bug-xsd-schema-not-foundFlow(bug-xsd-schema-not-foundFlow/processors/2 @ bug-xsd-schema-not-found-support:main.xml:16 (Validate schema))
+  (set debug level logging or '-Dmule.verbose.exceptions=true' for everything)
+********************************************************************************
+```
+
+This error occurs because the access was restricted in the XML module global element configuration by setting the *Expand entities* field to the default value of *NEVER*. This value prevents XXE (XML External Entity) and DoS (Denial of Service) attacks.
+
+To resolve this issue, set the *Expand entities* field to *INTERNAL*:
+
+. In Studio, open your project and access your Mule app flow.
+. Navigate to the *Global Element* tab.
+. Select the *XML Config* configuration and click *Edit*.
+. In the *Global Element Properties* window, set the *Expand entities* field to *INTERNAL*.
+. Click *OK*.
+
+image::xml-validate-schema-4.png[XML module global configuration Expand entities field]
+
+In the *Configuration XML* editor, the `<expandEntities>` configuration looks like this:
+
+[source,xml,linenums]
+----
+<xml-module:config name="XML_Config" expandEntities="INTERNAL"/>
+----
+
+
+== Understand Common Throws
+
+Here is a list of common throw messages and how to interpret them:
+
+* XML-MODULE:INVALID_INPUT_XML
+
+ The input document is not a valid XML.
+
+* XML-MODULE:INVALID_SCHEMA
+
+ The supplied schema is invalid.
+
+* XML-MODULE:INVALID_XPATH_EXPRESSION
+
+ The supplied XPath expression is invalid.
+
+* XML-MODULE:NULL_CONTEXT_PROPERTY
+
+ A context property with a null value was supplied.
+
+* XML-MODULE:SCHEMA_NOT_HONOURED
+
+ The input XML document does not honor its schema.
+
+ * XML-MODULE:SCHEMA_NOT_FOUND
+
+  The schema could not be found.
+
+* XML-MODULE:TRANSFORMATION
+
+ There was an error transforming an XML document.
+
+* XML-MODULE:SCHEMA_INPUT_ERROR
+
+There was an error when the Validate schema operation validated if both fields, Schema and Schema Content, are used at the same time at the execution time. Note that you can only use one of the fields to validate a schema XSD content.
```

Comment:
Maybe: The Validate schema operation failed because both the Schema and the Schema Content fields were used at the same time during the XML Module's execution. You can only use one of these fields to validate schema XSD content.